### DISCUSSION TITLE
Error: `redirect` can not be returned from getStaticProps during prerendering (/404)

### DISCUSSION THREAD

**Discussion by oran1248** [vercel/next.js#19033] [Help]
Hi,

I'm trying to redirect on 404 to main page I'm getting this error:
`Error: `redirect` can not be returned from getStaticProps during prerendering (/404)`

My `404.js` file content is:
```
export default function Custom404() {
	return null;
}

export const getStaticProps = async (ctx) => {
	return {
		props: {},
		redirect: {
			destination: '/',
			statusCode: 302,
		}
	};
};
```

It works only in development, but not when trying to build/deploy.

Please help

**Comment by kachkaev** [comment]
Instead of 404.js, you can try using an [optional catch-all route](https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes)

**Reply by oran1248** [reply]
Seems to me like a very strange workaround. 
So what `404.js`  is for?

**Reply by kachkaev** [reply]
404.js is for _showing_ Not found pages, automatically redirecting to `/` is generally undesired.

When I’ve somehow landed a wrong page, I want to see an "honest" response page with the URL left intact. This lets me fix the typo if I made any, which is not the case if the website 'wiped' my URL. Example:

https://github.com/kachkaev/njt/blob/main/src/pages/404.tsx
↓
https://njt.now.sh/oops

**Reply by oran1248** [reply]
But it's not strange to return 200 OK with response for 404 status?

**Reply by kachkaev** [reply]
`$ http https://njt.now.sh/oops`

```
HTTP/1.1 404 Not Found
Connection: keep-alive
Content-Encoding: gzip
Content-Type: text/html; charset=utf-8
Date: Wed, 11 Nov 2020 18:44:10 GMT
Transfer-Encoding: chunked
```

Next.js sets the right headers for you. If you have a page like `/products/[name].tsx`, you can return `notFound: true` in your `get*Props` function based on the URL. See https://nextjs.org/docs/basic-features/data-fetching → 🔎 `notFound`

**Reply by oran1248** [reply]
Oh cool it returns 404 with the content of `404.js`. cool! thanks

**Comment by Zebych** [comment]
To use redirects you can use the redirects key in next.config.js:

next.config.js

module.exports = {
  async redirects() {
    return [
      {
        source: '/about',
        destination: '/',
        permanent: true,
      },
    ]
  },
}
